intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
+
+\inputsite{available-modules}
+
+This gives a full list of software packages that can be loaded. Note that
+modules starting with a capital letter are listed first.
+
+\subsection{Organisation of modules in toolchains}
+
+The amount of modules on the VSC systems can be overwhelming, and it is not
+always immediately clear which modules can be loaded safely together if you
+need to combine multiple programs in a single job to get your work done.
+
+Therefore the VSC has defined so-called \strong{toolchains} on the newer VSC-clusters.
+A toolchain contains a C/C++ and Fortran compiler, a MPI library and some basic
+math libraries for (dense matrix) linear algebra and FFT. Two toolchains are
+defined on most VSC systems. One, the \lstinline|intel| toolchain, consists of the Intel
+compilers, MPI library and math libraries. The other one, the \lstinline|foss| toolchain,
+consists of Open Source components: the GNU compilers, OpenMPI, OpenBLAS and
+the standard LAPACK and ScaLAPACK libraries for the linear algebra operations
+and the FFTW library for FFT. The toolchains are refreshed twice a year,
+which is reflected in their name.
+
+E.g., \lstinline|foss-2014a| is the first version of the \lstinline|foss| toolchain in 2014.
+
+The toolchains are then used to compile a lot of the software installed on
+the VSC clusters. You can recognise those packages easily as they all contain the
+name of the toolchain after the version number in their name. Packages
+compiled with the same toolchain and toolchain version will typically work
+together well without conflicts.
+
+For some clusters, additional toolchains are defined, e.g., to take advantage
+of specific properties of that cluster such as GPU accelerators or special
+interconnect features that require a vendor-specific MPI-implementation.
+
+\subsection{Activating and de-activating modules}
+\label{subsec:activating-and-deactivating-modules}
+
+A module is loaded using the following command:
+
+
+\begin{prompt}
+%\shellcmd{module load \emph{examplemodule}}%
+\end{prompt}
+
+This will load the most recent version of \emph{examplemodule}.
+
+For some packages (e.g., OpenMPI), multiple versions are installed; the load
+command will automatically choose the most recent version (i.e., the
+lexicographical last after the ``/'') or the default version (as set by the
+system administrators). However, the user can (and probably should, to avoid
+surprises when newer versions are installed) specify a particular version,
+e.g.,
+
+\begin{prompt}
+%\shellcmd{module load \emph{secondexample/2.7-intel-2018a}}%
+\end{prompt}
+
+Obviously, you need to keep track of the modules that are currently
+loaded. If you executed the two load commands stated above, you will get the
+following:
+
+\begin{prompt}
+%\shellcmd{module list}%
+Currently Loaded Modulefiles:
+  1) examplemodule/1.2.3-intel-2018a        3) intel/2018a
+  2) secondexample/2.7-intel-2018a          4) examplelib/1.2-intel-2014b
```

Comment:
This can never happen, you can never load modules with two different `intel` versions.

Also, in reality you'll have over a dozen modules loaded as soon as you load something that also loads the `intel` toolchain...

Show that, and explain why you suddenly have a whole bunch of modules loaded.

This is a reason *not* to use `\year`, since the versions of the modules loaded by `intel` depend heavily on the `intel` version...